test/validate/validate.go

```diff
@@ -175,52 +207,72 @@ func (v *Validator) validateIPs(ctx context.Context, stateFileIps stateFileIpsFu
-		return errors.Wrapf(err, "failed to get node list")
-	}
+func validateNodeProperties(nodes *corev1.NodeList, labels map[string]string, ipCount int) error {
+	log.Print("Validating Node properties")
 
 	for index := range nodes.Items {
 		nodeName := nodes.Items[index].ObjectMeta.Name
 		// check nodes status;
 		// nodes status should be ready after cluster is created
 		nodeConditions := nodes.Items[index].Status.Conditions
 		if nodeConditions[len(nodeConditions)-1].Type != corev1.NodeReady {
-			return errors.Wrapf(err, "node %s status is not ready", nodeName)
+			return errors.Errorf("node %s status is not ready", nodeName)
 		}
 
 		// get node labels
 		nodeLabels := nodes.Items[index].ObjectMeta.GetLabels()
 		for key := range nodeLabels {
-			if label, ok := dualstackOverlayNodeLabels[key]; ok {
+			if label, ok := labels[key]; ok {
 				log.Printf("label %s is correctly shown on the node %+v", key, nodeName)
 				if label != overlayClusterLabelName {
-					return errors.Wrapf(err, "node %s overlay label name is wrong; expected label:%s but actual label:%s", nodeName, overlayClusterLabelName, label)
+					return errors.Errorf("node %s overlay label name is wrong; expected label:%s but actual label:%s", nodeName, overlayClusterLabelName, label)
 				}
 			}
 		}
 
-		// check if node has two internal IPs(one is IPv4 and another is IPv6)
+		// check if node has corret number two internal IPs
 		internalIPCount := 0
 		for _, address := range nodes.Items[index].Status.Addresses {
 			if address.Type == "InternalIP" {
 				internalIPCount++
 			}
 		}
-		if internalIPCount != 2 { //nolint
-			return errors.Wrap(err, "node does not have two internal IPs")
+		if internalIPCount != ipCount {
+			return errors.Errorf("node does not expected number of IPs %d", ipCount)
 		}
 	}
 
 	return nil
 }
 
 func (v *Validator) ValidateDualStackControlPlane(ctx context.Context) error {
-	if err := v.validateDualStackNodeProperties(ctx); err != nil {
+	nodes, err := k8sutils.GetNodeListByLabelSelector(ctx, v.clientset, nodeSelectorMap[v.os])
+	if err != nil {
+		return errors.Wrap(err, "failed to get node list")
+	}
+
+	if err := validateNodeProperties(nodes, dualstackOverlayNodeLabels, dualStackOverlayInternalIPs); err != nil {
 		return errors.Wrap(err, "failed to validate dualstack overlay node properties")
 	}
 
 	return nil
 }
+
+func (v *Validator) Validatev4OverlayControlPlane(ctx context.Context) error {
+	nodes, err := k8sutils.GetNodeListByLabelSelector(ctx, v.clientset, nodeSelectorMap[v.os])
+	if err != nil {
+		return errors.Wrap(err, "failed to get node list")
+	}
+
+	if err := validateNodeProperties(nodes, v4OverlayNodeLabels, ipv4OverlayInternalIPs); err != nil {
+		return errors.Wrap(err, "failed to validate IPv4 overlay node properties")
+	}
+
+	if v.os == "windows" {
+		if err := validateHNSNetworkState(ctx, nodes, v.clientset, v.config); err != nil {
+			return errors.Wrap(err, "failed to validate IPv4 overlay HNS network state")
+		}
```

Comment:
Is this an overlay exclusive HNS check? Maybe we should make a function that validates all HNS related items.